proxy/http/HttpConfig.cc

```diff
@@ -900,6 +900,68 @@ register_stat_callbacks()
                      (int)http_sm_finish_time_stat, RecRawStatSyncSum);
 }
 
+static bool
+set_negative_caching_list(const char *name, RecDataT dtype, RecData data, HttpConfigParams *c, bool update)
+{
+  bool ret = false;
+  // values from proxy.config.http.negative_caching_list
+  if (0 == strcasecmp("proxy.config.http.negative_caching_list", name) && RECD_STRING == dtype) {
+    if (data.rec_string) {
+      // parse the list of status code
+      ts::TextView status_list(std::string(data.rec_string));
+      while (!status_list.empty()) {
+        auto is_sep{[](char c) { return isspace(c) || ',' == c || ';' == c; }};
+        ts::TextView span, token{status_list.ltrim_if(is_sep).take_prefix_if(is_sep)};
+        auto n = ts::svtoi(token, &span);
+        if (span.size() != token.size()) {
+          Error("Invalid status code '%.*s' for negative caching: not a number", static_cast<int>(token.size()), token.data());
+        } else if (n <= 0 || n >= 600) {
+          Error("Invalid status code '%.*s' for negative caching: out of range", static_cast<int>(token.size()), token.data());
+        } else {
+          c->negative_caching_list[n] = true;
+        }
+      }
+      ret = true;
+    }
+  }
+  if (update == true) {
+    return ret;
+  }
+  return false;
+}
+
+// Method of getting the status code bitset
+static int
+negative_caching_list_cb(const char *name, RecDataT dtype, RecData data, void *cookie)
+{
+  HttpConfigParams *c = static_cast<HttpConfigParams *>(cookie);
+  // Signal an update if valid value arrived.
+  if (set_negative_caching_list(name, dtype, data, c, true)) {
+    http_config_cb(name, dtype, data, cookie);
+  }
+  return REC_ERR_OKAY;
+}
+
+void
+load_config_var(RecRecord const *r, void *cookie)
+{
+  HttpConfigParams *c = static_cast<HttpConfigParams *>(cookie);
+  set_negative_caching_list(r->name, r->data_type, r->data, c, false);
+
+  // The responses with the following status code WILL BE cached by default with negative caching enabled.
+  c->negative_caching_list[HTTP_STATUS_NO_CONTENT]            = true;
```

Comment:
Suppose I want to configure negative caching to not include `HTTP_STATUS_FORBIDDEN`. How do I do that?

Reply:
Yes, but in the previous logic they could not be configured at all. These should be used only if the incoming string is null or empty.